datatypes/src/util/any.rs

```diff
@@ -13,3 +13,20 @@ where
         self
     }
 }
+
+/// Easy `Any`-casting for `Arc`s by propagating the call to the underlying implementor
+pub trait AsAnyArc {
+    /// Returns the required Arc type for calling `Arc::downcast`
+    fn as_any_arc(self: Arc<Self>) -> Arc<(dyn Any + Send + Sync)>;
```

Comment:
producing an Arc only makes sense if we need more than one reference right? Do you really need that? Otherwise a &dyn Any would suffice, or would it not?

Reply:
We don't produce one. We consume one and basically just cast.
And I can't say self: `&Arc<Self>`, or can I?